### PR TITLE
refactor: various renaming/deobfusaction refactors

### PR DESCRIPTION
**Pull request by Jameskmonger**
I advise going commit-by-commit if you want to review this, it should be fairly easy to follow!

Tested and working on `server`

### CODE REVIEW

**Comment by MatthewBishop** [comment]
Ill check over this more later. I thought I had done a rename of LinkedList in an old PR. I guess I never submitted one. 

One thing that I would advise is

LinkedList.peekFront should be LinkedList.peekFirst instead to be in line with the naming of the java.util.collections.LinkedList.

**Comment by MatthewBishop** [comment]
method911 in LinkedList looks to be a method for insertion for a node arg2 prior to the node arg1. I would suggest naming this method as addBefore or insertBefore.

**Comment by Jameskmonger** [comment]
Thanks @MatthewBishop, I will go through and make those changes. I think I can remove most of the `_maybe`s tbh

File: src/main/java/org/runejs/client/Landscape.java [hunk 1]
```diff
@@ -96,9 +96,9 @@ public static void loadRegion() {
                         int offsetY = -Class26.baseY + (0xff & ISAAC.mapCoordinates[pointer]) * 64;
                         int offsetX = -MovedStatics.baseX + 64 * (ISAAC.mapCoordinates[pointer] >> 8);
                         byte[] is = RSString.terrainData[pointer];
-                        if(FileOperations.FileExists("./data/maps/" + LinkedList.anIntArray1071[pointer] + ".dat")) {
-                            System.out.println("reading file: " + "./data/maps/" + LinkedList.anIntArray1071[pointer] + ".dat");
-                            is = FileOperations.ReadFile("./data/maps/" + LinkedList.anIntArray1071[pointer] + ".dat");
+                        if(FileOperations.FileExists("./data/maps/" + LinkedList.terrainDataIds[pointer] + ".dat")) {
+                            System.out.println("reading file: " + "./data/maps/" + LinkedList.terrainDataIds[pointer] + ".dat");
```
Comment (gruckion) [inline src/main/java/org/runejs/client/Landscape.java:100]: Do we need this log in?

File: src/main/java/org/runejs/client/Class35.java [hunk 2]
```diff
@@ -11,118 +11,119 @@ public class Class35 implements Interface3 {
...
+        textures = new Texture[gameTextureCacheArchive.fileLength(0)];
+        for(int texturePointer = 0; textureCount > texturePointer; texturePointer++) {
+            Buffer textureBuffer = new Buffer(gameTextureCacheArchive.getFile(0, cacheTextures[texturePointer]));
+            textures[cacheTextures[texturePointer]] = new Texture(textureBuffer);
         }
 
     }
 
-    public static Class40_Sub5_Sub15 method421(CacheArchive arg0, byte arg1, int arg2, CacheArchive arg3, boolean arg4) {
+    public static Class40_Sub5_Sub15 method421(CacheArchive skeletonArchive, byte arg1, int arg2, CacheArchive skinArchive, boolean arg4) {
```
Comment (gruckion) [inline src/main/java/org/runejs/client/Class35.java:40]: What is skeletonArchive?

File: src/main/java/org/runejs/client/LinkedList.java [hunk 3]
```diff
@@ -66,113 +66,113 @@ else if(Buffer.anInt1985 != Player.worldLevel) {
...
+        return node;
     }
 
-    public Node method913(int arg0) {
-        if(arg0 != 25447)
-            method902((byte) -115);
-        Node class40 = aClass40_1056.next;
-        if(aClass40_1056 == class40)
+    public Node popFront(int arg0) {
+        Node next = tail.next;
```
Comment (gruckion) [inline src/main/java/org/runejs/client/LinkedList.java:170]: arg0 != 25447 was removed what was it doing?

Comment (Jameskmonger) [reply src/main/java/org/runejs/client/LinkedList.java:170]: just a dummy variable added by the obfuscator, was effectively unused (all code paths passed it in as `25447`)

File: src/main/java/org/runejs/client/Class35.java (hunk 2 above)
Comment (Jameskmonger) [reply src/main/java/org/runejs/client/Class35.java:40]: animation skeletons

https://www.rune-server.ee/runescape-development/rs2-server/informative-threads/700768-runescapes-animation-system.html

File: src/main/java/org/runejs/client/Landscape.java (hunk 1 above)
Comment (Jameskmonger) [reply src/main/java/org/runejs/client/Landscape.java:100]: I don't want to remove it here (was here before)

**Comment by Jameskmonger** [comment]
I just found and fixed an issue with this

I accidentally passed `hslBitset` as a parameter when I meant to pass `hslBitsetOriginal`

Resolved in 744ddf7

**before:**

image

**after:**

image

**Review by SchauweM — COMMENTED** [review]
@Kikorono @Promises can you verify?

**Pull request merged**